operator/api/v1beta1/lokistack_webhook_test.go

```diff
@@ -0,0 +1,139 @@
+package v1beta1_test
+
+import (
+	"testing"
+
+	"github.com/grafana/loki/operator/api/v1beta1"
+	"github.com/stretchr/testify/require"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+var ltt = []struct {
+	desc string
+	spec v1beta1.LokiStack
+	err  *apierrors.StatusError
+}{
+	{
+		desc: "valid spec",
+		spec: v1beta1.LokiStack{
+			Spec: v1beta1.LokiStackSpec{
+				Storage: v1beta1.ObjectStorageSpec{
+					Schemas: []v1beta1.ObjectStorageSchema{
+						{
+							Version:       v1beta1.ObjectStorageSchemaV11,
+							EffectiveDate: "2020-10-11",
+						},
+						{
+							Version:       v1beta1.ObjectStorageSchemaV12,
+							EffectiveDate: "2020-10-13",
+						},
+					},
+				},
+			},
+		},
+	},
+	{
+		desc: "not unique schema effective dates",
```

Comment:
For some reason this check does not work for me on the test-cluster. I am able to write a spec with two schema-configs with the same effective date. If they are in the right order (v11, v12, v11, for example) this will lead to a configuration that causes Loki to immediately fail to start up.

Reply:
Strange, the last time I tested this, it worked as expected (by that I mean that it blocked the change from occurring).